docs/inputs.rst

```diff
@@ -0,0 +1,215 @@
+combinations of values ``[]``, ``["a"]``, ``["b"]``, ``["c"]``,
+``["a", "b"]``, ``["a", "c"]``, ``["b", "c"]`` and ``["a", "b", "c"]``.
+
+This option is useful when you have a set of inputs that are optional
+and can be used with each other in any combination.
+
+
+Multiple input objects
+++++++++++++++++++++++
+
+So far each of the supported input options has been presented
+individually. You might naturally be wondering why inputs has been
+defined as a list of input objects.
+
+The reason for this is that an arbitrary number of input objects can
+be included in the inputs list. The implication of doing this is that
+all combinations of options are potentially valid inputs.
+
+If we combine two of the earlier examples into one ...
+::
+
+   >>> from melody.inputs import Fixed, Choice
+   >>> inputs = [FloatRange(name="range2", low=0.0, high=0.4, step=0.1),
+                 Choice(name="input2", inputs=["a", "b", "c"])]
+
+we will be specifying the following valid combinations for ``range2``
+and ``input2``: ``0.0, "a"``, ``0.0, "b"``, ``0.0, "c"``, ``0.1,
+"a"``, ``0.1, "b"``, ``0.1, "c"``, ``0.2, "a"``, ``0.2, "b"``, ``0.2,
+"c"``, ``0.3, "a"``, ``0.3, "b"`` and ``0.3, "c"``.
+
+.. note::
+
+   The last input object specified in the list iterates fastest,
+   followed by the penultimate one etc. So, in the above example the
+   values for ``input2`` are changing more rapidly than the values for
+   ``range2``.
+
+Extending
++++++++++
+
+If the supported input classes do not cover your requirements then you
+can create your own input classes. All of the input classes inherit
+from the Input base class.
+
+.. note::
+
+   Please ignore the state argument and method in the ``Input``
+   class. These are not used at the moment and are placeholders for
+   future developments.
+
+.. autoclass:: melody.inputs.Input
+    :members:
+
+You can subclass the input class. For example, if you wanted to
+provide all values greater than a tolerance as inputs from a list of values:
+::
+   
+   >>> from melody.inputs import Input
+   >>> class IntTolerance(Input):
+           ''' returns values if they are greater than a tolerance '''
+           def __init__(self, name, inputs, tolerance):
+               options = []
+	       for value in inputs:
+	           if value>tolerance:
+	               options.append(value)
+           Input.__init__(self, name, options)
+   >>> inputs = [IntTolerance("tolerance", [8, 9, 2, 4, 10], 7)]
+
+
+Alternatively you can subclass one of the supporting input types if
+that is simpler. For example, if you wanted to append a string to all
+Switch values:
+::
+
+   >>> from melody.inputs import Switch
+   >>> class SwitchAppend(Switch):
+           ''' append a string to all switch values '''
+           def __init__(self, name, off, on, append):
+               Switch.__init__(self, name, off+append, on+append)
+   >>> inputs = [SwitchAppend("switch", "a", "b", "_value")]
+
+Contributing
```

Comment:
I believe that contributing.md is to specify the rules of how to commit (coding rules, documentation, testing etc) and github will link to this in any pull request. I am simply asking for people to contribute rather than keeping their code locally. I agree that a contributing.md file would be useful, but I think that is a different point.